lib/ansible/module_utils/net_tools/nios/api.py

```diff
@@ -223,23 +217,32 @@ def run(self, ib_obj_type, ib_spec):
                 else:
                     proposed_object[key] = self.module.params[key]
 
-        modified = not self.compare_objects(current_object, proposed_object)
+        # checks if the name's field has been updated
+        if update and 'name' in proposed_object:
+            proposed_object['name'] = proposed_object['name']['new_name']
 
+        res = None
+        modified = not self.compare_objects(current_object, proposed_object)
         if 'extattrs' in proposed_object:
             proposed_object['extattrs'] = normalize_extattrs(proposed_object['extattrs'])
-
         if state == 'present':
             if ref is None:
                 if not self.module.check_mode:
                     self.create_object(ib_obj_type, proposed_object)
                 result['changed'] = True
             elif modified:
-                if 'network_view' in proposed_object:
-                    self.check_if_network_view_exists(proposed_object['network_view'])
+                if (ib_obj_type == NIOS_HOST_RECORD):
+                    proposed_object = self.on_update(proposed_object, ib_spec)
+                    res = self.update_object(ref, proposed_object)
+                elif (ib_obj_type == NIOS_NETWORK_VIEW):
+                    proposed_object = self.on_update(proposed_object, ib_spec)
+                    res = self.update_object(ref, proposed_object)
+                elif (ib_obj_type == NIOS_DNS_VIEW):
+                    proposed_object = self.on_update(proposed_object, ib_spec)
+                    res = self.update_object(ref, proposed_object)
+                elif 'network_view' in proposed_object:
```

Comment:
The if ..elif condition can be merged in single check

Reply:
Done